src/ossia-max/src/client.cpp

```diff
@@ -135,22 +144,29 @@ void* client::create(t_symbol* name, long argc, t_atom* argv)
 
 void client::destroy(client* x)
 {
+  auto pat_it = ossia_max::instance().patchers.find(x->m_patcher);
+  if(pat_it != ossia_max::instance().patchers.end())
+  {
+    auto& pat_desc = pat_it->second;
+    if(pat_desc.client == x)
+      pat_desc.client = nullptr;
+    if(pat_desc.empty())
+    {
+      ossia_max::instance().patchers.erase(pat_it);
+    }
+  }
+
   x->m_dead = true;
   x->m_matchers.clear();
 
-  clock_free((t_object*)x->m_clock);
-
-
-  // No more needed since all children
-  // are connected to node.about_to_be_deleted
-  // x->unregister_children();
+  if(x->m_clock)
+    clock_free((t_object*)x->m_clock);
 
   disconnect(x);
-  outlet_delete(x->m_dumpout);
+  if(x->m_dumpout)
+    outlet_delete(x->m_dumpout);
   ossia_max::instance().clients.remove_all(x);
-#if OSSIA_MAX_AUTOREGISTER
```

Comment:
retro-compatibility is hell :-)
anyway if it does, it will make more sense to add a global option to define this behaviour instead of a compiling flag
I'll do that if someone complains